stream-chat-android-client/src/main/java/io/getstream/chat/android/client/api/models/FilterObject.kt

```diff
@@ -0,0 +1,33 @@
+package io.getstream.chat.android.client.api.models
+
+public sealed class FilterObject
```

Comment:
Are we going to provide a way to create custom filters (for example searching for specific distinct channel: https://github.com/GetStream/stream-chat-android/blob/main/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/info/group/member/GroupChatInfoMemberOptionsViewModel.kt#L41)? If not, it would be great to double-check if we are not missing something

Reply:
This custom filter should be the same than:
```
Filters.and(
            Filters.eq("distinct", true),
            Filters.eq("members", listOf(memberId, chatDomain.currentUser.id))
        )
```

With the new Structure it will be:
```
AndFilterObject(
            listOf(
                EqualsFilterObject("distinct", true),
                EqualsFilterObject("members", listOf(memberId, chatDomain.currentUser.id))
            )
        )
```

We will update `Filters` class to use the new structure, so they will be able to use both way

Reply:
I think I tried to use two `Filters.eq` instead of custom `FilterObject` and then the query was failing, but we can check that later on.

Reply:
I will check before my next PR where I will remove old FilterObject class